### PR TITLE
Feature/databricks

### PR DESCRIPTION
**Pull request by fivetran-jamie**
Pull Request
**Are you a current Fivetran customer?** 

fivetran made PR

**What change(s) does this PR introduce?** 

- databricks compatibility
- merged https://github.com/fivetran/dbt_zendesk/pull/73

**Did you update the CHANGELOG?** 

- [x] Yes

**Does this PR introduce a breaking change?**

- [x] Yes (please provide breaking change details below.)
- [ ] No  (please provide an explanation as to how the change is non-breaking below.)
because of the changes in https://github.com/fivetran/dbt_zendesk/pull/73

**Did you update the dbt_project.yml files with the version upgrade (please leverage standard semantic versioning)? (In both your main project and integration_tests)** 

- [x] Yes

**Is this PR in response to a previously created Bug or Feature Request**

- [X] Yes, Issue/Feature #72 
- [ ] No 

**How did you test the PR changes?** 

 

- [x] CircleCi  
- [ ] Local (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 

- [ ] BigQuery
- [ ] Redshift
- [ ] Snowflake
- [ ] Postgres
- [x] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
🐦 

**Feedback**

We are so excited you decided to contribute to the Fivetran community dbt package! We continue to work to improve the packages and would greatly appreciate your [feedback](https://www.surveymonkey.com/r/DQ7K7WW) on our existing dbt packages or what you'd like to see next.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,90 +1,142 @@
...
-
-## Models
-
-This package contains transformation models, designed to work simultaneously with our [Zendesk Support source package](https://github.com/fivetran/dbt_zendesk_source). A dependency on the source package is declared in this package's `packages.yml` file, so it will automatically download when you run `dbt deps`. The primary outputs of this package are described below. Intermediate models are used to create these output models.
+<p align="center">
+    <a alt="License"
+        href="https://github.com/fivetran/dbt_zendesk/blob/main/LICENSE">
+        <img src="https://img.shields.io/badge/License-Apache%202.0-blue.svg" /></a>
+    <a alt="Fivetran-Release"
+        href="https://fivetran.com/docs/getting-started/core-concepts#releasephases">
```
Comment (fivetran-joemarkiewicz) [inline README.md:6]:
```suggestion
```
We can remove this badge.

File: README.md [hunk 2]
```diff
@@ -1,90 +1,142 @@
...
+### Databricks Dispatch Configuration
+If you are using a Databricks destination with this package you will need to add the below (or a variation of the below) dispatch configuration within your `dbt_project.yml`. This is required in order for the package to accurately search for macros within the `dbt-labs/spark_utils` then the `dbt-labs/dbt_utils` packages respectively.
+```yml
+dispatch:
+  - macro_namespace: dbt_utils
+    search_order: ['spark_utils', 'dbt_utils']
+```
+
+## Step 2: Install the package
+Include the following zendesk_source package version in your `packages.yml` file:
```
Comment (fivetran-joemarkiewicz) [inline README.md:55]:
```suggestion
Include the following zendesk package version in your `packages.yml` file:
```

File: README.md [hunk 3]
```diff
@@ -1,90 +1,142 @@
...
+
 ```
-## Package Maintenance
-The Fivetran team maintaining this package **only** maintains the latest version. We highly recommend you keep your `packages.yml` updated with the [dbt hub latest version](https://hub.getdbt.com/fivetran/zendesk/latest/). You may refer to the [CHANGELOG](/CHANGELOG.md) and release notes for more information on changes across versions.
+> **Note**: Do not include the Zendesk source package. The Zendesk transform package already has a dependency on the source in its own `packages.yml` file.
 
-## Configuration
-By default, this package looks for your Zendesk Support data in the `zendesk` schema of your [target database](https://docs.getdbt.com/docs/running-a-dbt-project/using-the-command-line-interface/configure-your-profile). If this is not where your Zendesk Support data is, add the following configuration to your `dbt_project.yml` file:
+## Step 3: Define database and schema variables
+By default, this package runs using your destination and the `zendesk` schema. If this is not where your zendesk data is (for example, if your zendesk schema is named `zendesk_fivetran`), add the following configuration to your root `dbt_project.yml` file:
```
Comment (fivetran-joemarkiewicz) [inline README.md:66]: I think we can make this a bit clearer.
```suggestion
By default, this package runs using your destination and the `zendesk` schema. If this is not where your zendesk data is (for example, if your zendesk schema is named `zendesk_fivetran`), update the following variables in your root `dbt_project.yml` file accordingly:
```

File: README.md [hunk 4]
```diff
@@ -1,90 +1,142 @@
...
 ...
+config-version: 2
+
+vars:
+  zendesk:
+    ticket_field_history_timeframe_years: integer_number_of_years # default = 50 (everything)
+```
+
+### Changing the Build Schema
+By default this package will build the Zendesk staging models within a schema titled (<target_schema> + `_zendesk_staging`), the Zendesk intermediate models within a schema titled (<target_schema> + `_zendesk_intermediate`), and the Zendesk final models within a schema titled (<target_schema> + `_zendesk`) in your target database. If this is not where you would like your modeled Zendesk data to be written to, add the following configuration to your root `dbt_project.yml` file:
```
Comment (fivetran-joemarkiewicz) [inline README.md:137]: We should update this to reflect the updates from the source package. In particular we should adjust `_zendesk_staging` to instead be `_zendesk_source`.

File: README.md [hunk 5]
```diff
@@ -96,76 +148,54 @@ models:
...
+Fivetran offers the ability for you to orchestrate your dbt project through [Fivetran Transformations for dbt Core™](https://fivetran.com/docs/transformations/dbt). Learn how to set up your project for orchestration through Fivetran in our [Transformations for dbt Core setup guides](https://fivetran.com/docs/transformations/dbt#setupguide).
+</details>
+
+# 🔍 Does this package have dependencies?
+This dbt package is dependent on the following dbt packages. Please be aware that these dependencies are installed by default within this package. For more information on the following packages, refer to the [dbt hub](https://hub.getdbt.com/) site.
+> IMPORTANT: If you have any of these dependent packages in your own `packages.yml` file, we highly recommend that you remove them from your root `packages.yml` to avoid package version conflicts.
+    
 ```yml
-# dbt_project.yml
+packages:
```
Comment (fivetran-joemarkiewicz) [inline README.md:176]: We should also add spark utils here now.

**Review by fivetran-joemarkiewicz — CHANGES_REQUESTED** [review]
@fivetran-jamie thanks for applying the databricks updates and also merging the package updates PR! 🙌 

I have a few comments (mainly CHANGELOG and README related) that we should update before approval. Let me know if you have any questions!

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,3 +1,10 @@
+# dbt_zendesk_source v0.9.1
+## 🎉 Documentation and Feature Updates
+- Databricks compatibility 🧱 ([#74](https://github.com/fivetran/dbt_zendesk/pull/74))
+- README clarifications on installing the source and transform Zendesk packages together.
+- Updated README documentation updates for easier navigation and setup of the dbt package
+- Added `zendesk_[source_table_name]_identifier` variables to allow for easier flexibility of the package to refer to source tables with different names.
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:6]: consolidated and added

File: CHANGELOG.md [hunk 7]
```diff
@@ -1,3 +1,10 @@
+# dbt_zendesk_source v0.9.1
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:1]: ah yes

File: models/agent_work_time/int_zendesk__ticket_work_time_business.sql [hunk 8]
```diff
@@ -33,18 +33,24 @@ with ticket_historical_status as (
...
     select 
-      ticket_status_crossed_with_schedule.*,
-    ({{ fivetran_utils.timestamp_diff(
+      ticket_id,
+      ticket_status,
+      schedule_id,
+      status_schedule_start,
+      status_schedule_end,
+      status_valid_starting_at,
+      status_valid_ending_at,
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply models/agent_work_time/int_zendesk__ticket_work_time_business.sql:47]: yeah you can't `select *` and use a `group by` on spark, so i had to explicitly select the fields

File: models/sla_policy/agent_work_time/int_zendesk__agent_work_time_business_hours.sql [hunk 9]
```diff
@@ -41,25 +41,35 @@ with agent_work_time_filtered_statuses as (
...
+      {# ticket_status_crossed_with_schedule.*, --here #}
+      ticket_id,
+      sla_applied_at,
+      target,    
+      sla_policy_name,    
+      schedule_id,
+      valid_starting_at,
+      valid_ending_at,
+      status_valid_starting_at,
+      status_valid_ending_at,
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply models/sla_policy/agent_work_time/int_zendesk__agent_work_time_business_hours.sql:61]: same reason as above, but will remove the leftover comment!

File: models/sla_policy/requester_wait_time/int_zendesk__requester_wait_time_business_hours.sql [hunk 10]
```diff
@@ -41,25 +41,34 @@ with requester_wait_time_filtered_statuses as (
...
+      ticket_id,
+      sla_applied_at,
+      target,
+      sla_policy_name,
+      schedule_id,
+      valid_starting_at,
+      valid_ending_at,
+      status_valid_starting_at,
+      status_valid_ending_at,
+      ({{ dbt_utils.datediff(
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply models/sla_policy/requester_wait_time/int_zendesk__requester_wait_time_business_hours.sql:61]: same `*/group by` issue

File: CHANGELOG.md [hunk 11]
```diff
@@ -1,3 +1,10 @@
...
+## 🎉 Documentation and Feature Updates
+- Databricks compatibility 🧱 ([#74](https://github.com/fivetran/dbt_zendesk/pull/74))
+- README clarifications on installing the source and transform Zendesk packages together.
+- Updated README documentation updates for easier navigation and setup of the dbt package
+- Added `zendesk_[source_table_name]_identifier` variables to allow for easier flexibility of the package to refer to source tables with different names.
+
```
[the remark this answers is not in the document]
Comment (fivetran-jamie) [reply CHANGELOG.md:7]: good thinking

File: README.md (hunk 4 above)
Comment (fivetran-jamie) [reply README.md:135]: updated

File: README.md (hunk 5 above)
Comment (fivetran-jamie) [reply README.md:174]: added here and to the source README

File: CHANGELOG.md [hunk 12]
```diff
@@ -1,3 +1,12 @@
+# dbt_zendesk v0.9.0
```
Comment (fivetran-joemarkiewicz) [inline CHANGELOG.md:1]: I feel we should also add the `_zendesk_source` schema change notes here as well. Just so people that notice the change can see all relevant notes here.

File: models/agent_work_time/int_zendesk__ticket_work_time_business.sql (hunk 8 above)
[the remark this answers is not in the document]
Comment (fivetran-joemarkiewicz) [reply models/agent_work_time/int_zendesk__ticket_work_time_business.sql:47]: Ahhh that makes a lot of sense. Thanks for the clarification!

**Review by fivetran-joemarkiewicz — APPROVED** [review]
Looks great @fivetran-jamie! Just have a request to add another item to the CHANGELOG for the `_zendesk_source` default schema change.

Other than that, this looks good to go!

**Pull request merged**